File: SDKs/Aspose.Cells_Cloud_SDK_For_CSharp/src/Com/Aspose/cells/Model/Workbook.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.Aspose.Cells.Model {
  public class Workbook {
    public string FileName { get; set; }

    public List<Link> Links { get; set; }

    public LinkElement Worksheets { get; set; }

    public LinkElement DefaultStyle { get; set; }

    public LinkElement DocumentProperties { get; set; }

    public LinkElement Names { get; set; }

    public LinkElement Settings { get; set; }

    public string IsWriteProtected { get; set; }

    public string IsProtected { get; set; }

    public string IsEncryption { get; set; }

    public string Password { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class Workbook {\n");
      sb.Append("  FileName: ").Append(FileName).Append("\n");
      sb.Append("  Links: ").Append(Links).Append("\n");
      sb.Append("  Worksheets: ").Append(Worksheets).Append("\n");
      sb.Append("  DefaultStyle: ").Append(DefaultStyle).Append("\n");
      sb.Append("  DocumentProperties: ").Append(DocumentProperties).Append("\n");
      sb.Append("  Names: ").Append(Names).Append("\n");
      sb.Append("  Settings: ").Append(Settings).Append("\n");
      sb.Append("  IsWriteProtected: ").Append(IsWriteProtected).Append("\n");
      sb.Append("  IsProtected: ").Append(IsProtected).Append("\n");
      sb.Append("  IsEncryption: ").Append(IsEncryption).Append("\n");
      sb.Append("  Password: ").Append(Password).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }

File: SDKs/Aspose.Cells-Cloud-SDK-For-Ruby/lib/aspose_cells_cloud/models/workbook.rb
module AsposeCellsCloud
  # 
  class Workbook < BaseObject
    attr_accessor :file_name, :links, :worksheets, :default_style, :document_properties, :names, :settings, :is_write_protected, :is_protected, :is_encryption, :password
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # 
        :'file_name' => :'FileName',
        
        # 
        :'links' => :'Links',
        
        # 
        :'worksheets' => :'Worksheets',
        
        # 
        :'default_style' => :'DefaultStyle',
        
        # 
        :'document_properties' => :'DocumentProperties',
        
        # 
        :'names' => :'Names',
        
        # 
        :'settings' => :'Settings',
        
        # 
        :'is_write_protected' => :'IsWriteProtected',
        
        # 
        :'is_protected' => :'IsProtected',
        
        # 
        :'is_encryption' => :'IsEncryption',
        
        # 
        :'password' => :'Password'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'file_name' => :'String',
        :'links' => :'Array<Link>',
        :'worksheets' => :'LinkElement',
        :'default_style' => :'LinkElement',
        :'document_properties' => :'LinkElement',
        :'names' => :'LinkElement',
        :'settings' => :'LinkElement',
        :'is_write_protected' => :'String',
        :'is_protected' => :'String',
        :'is_encryption' => :'String',
        :'password' => :'String'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'FileName']
        self.file_name = attributes[:'FileName']
      end
      
      if attributes[:'Links']
        if (value = attributes[:'Links']).is_a?(Array)
          self.links = value
        end
      end
      
      if attributes[:'Worksheets']
        self.worksheets = attributes[:'Worksheets']
      end
      
      if attributes[:'DefaultStyle']
        self.default_style = attributes[:'DefaultStyle']
      end
      
      if attributes[:'DocumentProperties']
        self.document_properties = attributes[:'DocumentProperties']
      end
      
      if attributes[:'Names']
        self.names = attributes[:'Names']
      end
      
      if attributes[:'Settings']
        self.settings = attributes[:'Settings']
      end
      
      if attributes[:'IsWriteProtected']
        self.is_write_protected = attributes[:'IsWriteProtected']
      end
      
      if attributes[:'IsProtected']
        self.is_protected = attributes[:'IsProtected']
      end
      
      if attributes[:'IsEncryption']
        self.is_encryption = attributes[:'IsEncryption']
      end
      
      if attributes[:'Password']
        self.password = attributes[:'Password']
      end
      
    end

  end
end
